news/6305.bugfix

```diff
@@ -0,0 +1 @@
+When you update a block's properties using `onChangeBlock`, the widget may not show these changes because its state isn't updating correctly. @tedw87
```

Comment:
This is a backport of https://github.com/plone/volto/pull/6036. So please use the same text for the release notes.

```suggestion
Fix internalUrl Widget to Reflect Prop Changes via onChangeBlock @dobri1408 @ichim-david @tedw87
```